tensorflow/core/kernels/sparse_matmul_op.h

```diff
@@ -40,6 +40,36 @@ EIGEN_DEVICE_FUNC inline Packet pexpand_bf16_u(const Packet& from) {
   return reinterpret_cast<const float&>(tmp);
 }
 
+// Specialization non-scalar version on non-sse.
+#ifndef EIGEN_VECTORIZE_SSE2
+template <typename Packet>
+EIGEN_DEVICE_FUNC inline Packet4f pexpand_bf16_l(const Packet4f& from) {
+  Packet4f res;
```

Comment:
The packet4f struct should remain opaque. A more portable version of the code would be:
float r[4];
pstoreu(r, from);
float p[4];
 p[0] = static_cast<float>((r[2] << 16) & 0xffff0000);
 p[1] = static_cast<float>(r[2] & 0xffff0000);
 p[2] = static_cast<float>((r[3] << 16) & 0xffff0000);
 p[3] = static_cast<float>(r[3] & 0xffff0000);
return ploadu(p);